massa-network/src/peer_info_database.rs

```diff
@@ -443,6 +443,119 @@ impl PeerInfoDatabase {
+                    )?
+                }
+                if peer.active_out_connections > 0 {
+                    self.update_global_active_out_connection_count(
+                        old_pt,
+                        false,
+                        NetworkConnectionErrorType::TooManyConnectionAttempt(ip),
+                    )?;
+                    self.update_global_active_out_connection_count(
+                        PeerType::WhiteListed,
+                        true,
+                        NetworkConnectionErrorType::UnexpectedError,
+                    )?
+                }
+                if peer.active_in_connections > 0 {
+                    self.update_global_active_in_connection_count(
+                        old_pt,
+                        false,
+                        NetworkConnectionErrorType::TooManyConnectionAttempt(ip),
+                    )?;
+                    self.update_global_active_in_connection_count(
+                        PeerType::WhiteListed,
+                        true,
+                        NetworkConnectionErrorType::UnexpectedError,
+                    )?
+                }
+            }
+        }
+        self.update()
+    }
+
+    pub async fn remove_from_whitelist(&mut self, ips: Vec<IpAddr>) -> Result<(), NetworkError> {
+        for ip in ips.into_iter() {
+            let old_pt = if let Some(peer) = self.peers.get_mut(&ip) {
+                let old = peer.peer_type;
+                peer.peer_type = Default::default();
+                old
+            } else {
+                return Ok(());
+            };
+
+            if old_pt != Default::default() {
+                // update global connection counts by peer type
+                let peer = *self.peers.get(&ip).unwrap(); // filled just before
+                if peer.active_out_connection_attempts > 0 {
+                    self.update_global_active_out_connection_attempt_count(
+                        old_pt,
+                        false,
+                        NetworkConnectionErrorType::TooManyConnectionAttempt(ip),
+                    )?;
+                    self.update_global_active_out_connection_attempt_count(
+                        Default::default(),
+                        true,
+                        NetworkConnectionErrorType::UnexpectedError,
+                    )?
+                }
+                if peer.active_out_connections > 0 {
+                    self.update_global_active_out_connection_count(
+                        old_pt,
+                        false,
+                        NetworkConnectionErrorType::TooManyConnectionAttempt(ip),
+                    )?;
+                    self.update_global_active_out_connection_count(
+                        Default::default(),
+                        true,
+                        NetworkConnectionErrorType::UnexpectedError,
+                    )?
+                }
```

Comment:
It can be factorized in a small function or in a loop. Can you use a better error for the update?

Reply:
it's not exactly the same, so I don't see how :bike: 

I have no name idea for the errors